src/extensions/surveys.tsx

```diff
@@ -219,3 +118,752 @@ export function generateSurveys(posthog: PostHog) {
         callSurveys(posthog, false)
     }, 3000)
 }
+
+const defaultBackgroundColor = '#eeeded'
+
+export function QuestionHeader({
+    question,
+    description,
+    backgroundColor,
+}: {
+    question: string
+    description?: string | null
+    backgroundColor?: string
+}) {
+    return (
+        <div style={{ backgroundColor: backgroundColor || defaultBackgroundColor }}>
+            <div className="survey-question">{question}</div>
+            {description && <div className="description" dangerouslySetInnerHTML={{ __html: description }} />}
+        </div>
+    )
+}
+
+export function Cancel({ onClick }: { onClick: () => void }) {
+    return (
+        <div className="cancel-btn-wrapper">
+            <button className="form-cancel" onClick={onClick}>
+                {cancelSVG}
+            </button>
+        </div>
+    )
+}
+
+export function BottomSection({
+    text,
+    submitDisabled,
+    appearance,
+    onSubmit,
+    link,
+}: {
+    text: string
+    submitDisabled: boolean
+    appearance: SurveyAppearance
+    onSubmit: () => void
+    link?: string | null
+}) {
+    const backgroundColorRef = Preact.createRef()
```

Comment:
Why is this `createRef`? I would have assumed `useRef()` would be the functional way of working? Otherwise that create call is happening on every render which means it doesn't feed into the hook system. 

Same applies for all uses of it I can see 🤔 

Also the ref is to the button not the color

Reply:
```suggestion
    const buttonRef = useRef<HTMLButtonElement>(null)
```